WebApp/src/main/java/com/isa/webapp/controller/TeacherController.java

```diff
@@ -1,115 +1,80 @@
-                             @AuthenticationPrincipal UserDetails userDetails) {
-        boolean isTeacher = userDetails.getAuthorities().stream()
-                .anyMatch(a -> a.getAuthority().equals("TEACHER"));
-        Map<Subjects, List<Integer>> grades = studentService.getGradesForStudent(studentId);
+    @GetMapping("/teacher/view-grades/{studentId}")//todo
+    public String viewGrades(@PathVariable String studentId, Model model, @AuthenticationPrincipal UserDetails userDetails) {
+        boolean isTeacher = userDetails.getAuthorities().stream().anyMatch(a -> a.getAuthority().equals("TEACHER"));
+        List<Grade> studentGrades = gradeRepository.findByUserUuid(studentId);
+        Map<Subject, List<Double>> grades = convertGradesToMap(studentGrades);
         model.addAttribute("grades", grades);
         model.addAttribute("isTeacher", isTeacher); //TODO fix
         return "teacher_view_grades";
     }
 
     @PostMapping("/teacher/add-grade")
-    public String addGrade(@ModelAttribute("gradeForm") GradeForm gradeForm) {
-        addGradeToStudent(gradeForm.getStudentId(), gradeForm.getSubject(), gradeForm.getGrade());
+    public String addGrade(@RequestParam("studentId") String studentUuid, @RequestParam("subject") Subject subject, @RequestParam("grade") Double grade, @AuthenticationPrincipal UserDetails userDetails) {
+        User user = userRepository.findByUuid(studentUuid).orElse(null);
+        Grade newGrade = new Grade();
+        newGrade.setSubjects(subject);
+        newGrade.setValue(grade);
+        newGrade.setDate(LocalDate.now());
+        newGrade.setUser(user);
+        gradeRepository.save(newGrade);
         return "redirect:/teacher/students";
     }
 
-    public List<User> getStudentsForTeacher() throws IOException {
-        ObjectMapper mapper = new ObjectMapper();
-        List<User> users = mapper.readValue(Files.newBufferedReader(Paths.get(USERS_JSON_FILE)), new TypeReference<>() {});
-
-        return users.stream()
-                .filter(user -> user.getUserRole() == UserRole.STUDENT)
-                .collect(Collectors.toList());
-    }
-
-    public void addGradeToStudent(String studentId, Subjects subject, int grade) {
-        try {
-            ObjectMapper mapper = new ObjectMapper();
-            List<User> users = mapper.readValue(Files.newBufferedReader(Paths.get(USERS_JSON_FILE)), new TypeReference<>() {});
-
-            for (User user : users) {
-                if (user.getId().equals(studentId)) {
-                    Map<Subjects, List<Integer>> grades = user.getGrades();
-                    grades.computeIfAbsent(subject, key -> new ArrayList<>()).add(grade);
-                    Files.write(Paths.get(USERS_JSON_FILE), mapper.writeValueAsBytes(users));
-                    break;
-                }
-            }
-        } catch (IOException e) {
-            e.printStackTrace();
-        }
+    public Map<Subject, List<Double>> convertGradesToMap(List<Grade> grades) {
```

Comment:
to do serwisu można

Reply:
do serwisu jako normalna metoda a nie statyczna, skoro mamy zależność i tak, więc dlaczego z niej nie skorzystać, a po drugie, dobrze by było zrobić jedną metodę, np. `getGrades()`, która przyjmuje `studentId` i w jednej metodzie wyciągasz z bazy i od razu konwertujesz do mapy